### PR TITLE
[BD-21] Add featuretoggle configuration and sphinx extension

### PR DESCRIPTION
**Pull request by regisb**
**Description:** We introduce a new Sphinx extension for generating human-readable
documentation of feature toggle annotations. This extension introduces a
.. featuretoggles:: directive that will list feature toggles for the
configured code base. Annotations are collected on-the-fly using
the configuration file from the config_and_tools folder.

JIRA: https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596358943

Dependencies: related edx-platform PR: edx/edx-platform#24344

**Reviewers:**
- [x] @robrap 

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** 

* The duplicate "feature_toggle_annotations.yaml" file should be removed from the edx-toggles repository.
* My feeling is that it is cumbersome to write `extensions = ["code_annotations.config_and_tools.sphinx.extensions.featuretoggles"]`. I propose that the "config_and_tools" folder be renamed to "contrib". More information on the "contrib" folder name:

    - https://softwareengineering.stackexchange.com/questions/252053/whats-in-the-contrib-folder
    - https://blog.startifact.com/posts/against-contrib.html

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @regisb! I've created [BLENDED-536](https://openedx.atlassian.net/browse/BLENDED-536) to keep track of it in Jira.   More details are on the [BD-21](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596358943/BD-21+Toggles+Settings+Documentation) project page.

When this pull request is ready, tag your edX technical lead.

**Comment by regisb** [comment]
@robrap This is ready for review.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -11,9 +11,11 @@ Change Log
...
 .. There should always be an "Unreleased" section for changes pending release.
 
-Unreleased
-~~~~~~~~~~
+[0.5.0] - 2020-08-06
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+* Add ``featuretoggles`` Sphinx extension
+* Include ``config_and_tools`` folder in pip-installable package
 * Add ADR 0001-config-and-tools.rst for adding a placing in this repository for shared annotation configs and supporting tools.
```
Comment (robrap) [inline CHANGELOG.rst:19]: Would you mind fixing my typo? `adding a placing` -> `adding a place`

File: code_annotations/config_and_tools/config/feature_toggle_annotations.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    - ".. documented_elsewhere_name:":
+  feature_toggle:
+    - ".. toggle_name:":
+    - ".. toggle_implementation:":
+        choices: [ExperimentWaffleFlag, WaffleFlag, WaffleSample, WaffleSwitch, CourseWaffleFlag, ConfigurationModel, DjangoSetting]
+    - ".. toggle_default:":
+    - ".. toggle_description:":
+    - ".. toggle_category:":
+    - ".. toggle_use_cases:":
+        choices: [incremental_release, launch_date, monitored_rollout, graceful_degradation, beta_testing, vip, opt_out, opt_in, open_edx]
```
Comment (robrap) [inline code_annotations/config_and_tools/config/feature_toggle_annotations.yaml:19]: Not as a blocker for this PR, but we should discuss what is really necessary here at some point.

**Comment by robrap** [comment]
@regisb: This has been tested.  I can merge and publish once you fix the typo and squash.  Thanks.

**Comment by regisb** [comment]
> I can merge and publish once you fix the typo and squash

I fixed the typo. Should I squash all the commits? It seems to me that they make more sense individually when they are separated. But if necessary I can merge them.

**Comment by robrap** [comment]
I was just requesting that you squash whatever you intend to squash. If you are happy as is, I can merge.

**Comment by regisb** [comment]
Sure! I already squashed the commits, so you can merge 😊

**Pull request merged**